job/external_dependency_resolver.go

```diff
@@ -65,11 +62,8 @@ func (e *externalDependencyResolver) FetchStaticExternalDependenciesPerJobName(c
 	var unknownDependencies []models.UnknownDependency
 	externalDependencyPerJobName := make(map[string]models.ExternalDependency)
 	for jobName, toBeResolvedDependency := range unresolvedDependenciesPerJobName {
-		optimusDependencies, unresolvedFromExternal, err := e.fetchStaticOptimusDependencies(ctx, toBeResolvedDependency)
-		if err != nil {
-			return nil, nil, err
-		}
-		unknownDependencies = convertUnresolvedToUnknownDependencies(jobName, unresolvedFromExternal)
+		optimusDependencies, unresolvedFromExternal := e.fetchStaticOptimusDependencies(ctx, toBeResolvedDependency)
+		unknownDependencies = e.convertUnresolvedToUnknownDependencies(jobName, unresolvedFromExternal)
```

Comment:
@irainia let's append this `unknownDependencies` so it will not be replaced in every loop? looks like this is missed in the previous commit.

Reply:
apart from this, looks good.

Reply:
updated